sdk/nodejs/output.ts

```diff
@@ -114,6 +114,52 @@ export class Output<T> {
         return obj && obj.__pulumiOutput ? true : false;
     }
 
+    /**
+     * [toString] on an [Output<T>] is not supported.  This is because the value an [Output] points
+     * to is asynchronously computed (and thus, this is akin to calling [toString] on a [Promise]).
+     * As such, this method only throws if called.
+     *
+     * To get the value of an Output<T> as an Output<string> consider either:
+     * 1. `o.apply(v => ``prefix${v}suffix``)` or
+     * 2. `pulumi.interpolate ``prefix${v}suffix`` `
+     *
+     * This will return an Output with the inner computed value and all resources still tracked.
+     */
+    public toString(): string {
+        // Note: i've mixed/matched `` and "" strings because I could not find a way to
+        // properly embed literals with `` inside another `` string.
```

Comment:
You can't escape `?

Reply:
i tried everything i coudl think of, and kept running into problems.  note: i think i got TS working at some point, but VSCode then went off the raise in terms of colorization.  given hte importance of this file, i didn't want that :)

Reply:
Ok.  was dumb.  figured it out.